catalog/urls.py

```diff
@@ -13,4 +13,14 @@
         name="product_detail",
     ),
     path("add_product/", views.ProductCreateView.as_view(), name="product_create"),
+    path(
+        "product_detail/<int:pk>/update",
```

Comment:
Забыл слеш в конце, ниже тоже
```suggestion
        "product_detail/<int:pk>/update/",
```